tests/test_cupy_compatibility.py

```diff
@@ -0,0 +1,20 @@
+def test_cupy_compatibility():
+    try:
+        import cupy
+    except ImportError:
+        # cupy may not be installed in the CI
+        return
```

Comment:
```suggestion
    import pytest
    cupy = pytest.importorskip('cupy')
```

(preferable because it will then show as skipped and not "passed")
https://docs.pytest.org/en/6.2.x/reference.html#pytest-importorskip